public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -511,7 +511,13 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
       const tags: string[] = [];
       for (const field of frame.fields) {
         if (field.labels) {
-          tags.push.apply(tags, [...new Set(Object.values(field.labels).map((label: string) => label.trim()))]);
+          tags.push.apply(tags, [
+            ...new Set(
+              Object.values(field.labels)
+                .filter((label: string) => label !== '')
```

Comment:
Question: Would it be worth to swap `filter` and `map` to filter out labels containing only white-space characters?

Reply:
Ah good call, yep! I added a test case for that too.